src/HistoryDialog.vala

```diff
@@ -114,7 +119,12 @@ namespace PantheonCalculator {
         }
 
         private void on_response (Gtk.Dialog source, int response_id) {
-            if (response_id == Gtk.ResponseType.OK) {
+            if (response_id == -1) {
```

Comment:
It is common to use a `switch {}` clause to deal with multiple responses.  At least using `if ... else if ` rather than two separate if statements is better style.